pkg/cluster/service_test.go

```diff
@@ -58,3 +65,95 @@ func TestDeleteToolchainClusterWhenDoesNotExist(t *testing.T) {
 		return nil
 	})
 }
+
+func TestListToolchainClusterConfigs(t *testing.T) {
+	// given
+	status := test.NewClusterStatus(toolchainv1alpha1.ToolchainClusterReady, corev1.ConditionTrue)
+	m1, sec1 := test.NewToolchainClusterWithEndpoint("east", "secret1", "http://m1.com", status, verify.Labels(cluster.Member, test.MemberOperatorNs, "m1ClusterName"))
+	m2, sec2 := test.NewToolchainClusterWithEndpoint("west", "secret2", "http://m2.com", status, verify.Labels(cluster.Member, test.MemberOperatorNs, "m2ClusterName"))
+	host, secHost := test.NewToolchainCluster("host", "secretHost", status, verify.Labels(cluster.Host, test.HostOperatorNs, "hostClusterName"))
+	noise, secNoise := test.NewToolchainCluster("noise", "secretNoise", status, verify.Labels(cluster.Type("e2e"), test.MemberOperatorNs, "noiseClusterName"))
+	require.NoError(t, toolchainv1alpha1.AddToScheme(scheme.Scheme))
+	cl := test.NewFakeClient(t, m1, m2, host, noise, sec1, sec2, secHost, secNoise)
+
+	t.Run("list members", func(t *testing.T) {
+		// when
+		clusterConfigs, err := cluster.ListToolchainClusterConfigs(cl, m1.Namespace, cluster.Member, time.Second)
+
+		// then
+		require.NoError(t, err)
+		require.Len(t, clusterConfigs, 2)
+		verify.AssertClusterConfigThat(t, clusterConfigs[0]).
+			IsOfType(cluster.Member).
+			HasName("east").
+			HasOperatorNamespace("toolchain-member-operator").
+			HasOwnerClusterName("m1ClusterName").
+			HasAPIEndpoint("http://m1.com").
+			RestConfigHasHost("http://m1.com")
+		verify.AssertClusterConfigThat(t, clusterConfigs[1]).
+			IsOfType(cluster.Member).
+			HasName("west").
+			HasOperatorNamespace("toolchain-member-operator").
+			HasOwnerClusterName("m2ClusterName").
+			HasAPIEndpoint("http://m2.com").
+			RestConfigHasHost("http://m2.com")
+	})
+
+	t.Run("list host", func(t *testing.T) {
+		// when
+		clusterConfigs, err := cluster.ListToolchainClusterConfigs(cl, m1.Namespace, cluster.Host, time.Second)
```

Comment:
```suggestion
		clusterConfigs, err := cluster.ListToolchainClusterConfigs(cl, host.Namespace, cluster.Host, time.Second)
```

Reply:
this code lists `ToolchainClusters` with the type `host` in the member namespace - in other words, this returns the list of host clusters registered for the particular member cluster. The assertion `HasOperatorNamespace` then verifies that the operator namespace of the found `ToolchianCluster` CR is really the expected host operator namespace.

Reply:
ok